File: iotOnGolang/internal/model/Room.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Room Declaration
//==============================================================
type Room struct {
    gorm.Model
     Name                                    string
    FloorId         *uint
    Floor           *Floor `gorm:"foreignKey:FloorId"`
     Devices           []IoTDevice `gorm:"foreignKey:DevicesFromRoomId"`
     Gateways           []Gateway `gorm:"foreignKey:GatewaysFromRoomId"`

// parent associations as their child

}

File: iotOnDjango/models/Room.py

from django.db import models

#======================================================================
# Class Room Declaration
#======================================================================
class Room (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	floor = models.ForeignKey('Floor', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	devices = models.ManyToManyField('IoTDevice',  blank=True, related_name='+')
	gateways = models.ManyToManyField('Gateway',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Room";
    
	def objectType(self):
		return "Room";
